internal/models/model_constants.go

```diff
@@ -31,6 +34,8 @@ func ProviderTypeFromString(str string) ProviderType {
 		return ProviderTypeAzure
 	case "gcp":
 		return ProviderTypeGCP
+	case "":
+		return ProviderTypeAll
 	default:
 		return ProviderTypeUnknown
 	}
```

Comment:
I don’t like introducing a constant that is very similar if not the exactly the same as `ProviderTypeUnknown`. I created it exactly for this purpose, if you don’t like the name then rename it. Maybe `ProviverTypeUnspecified` is more appropriate for your use case.

Reply:
But my problem is not the naming, there are two cases here which I want to differentiate: 
- If we get an empty provider we should call  `ListAllProvisioningSources`
-  If the provider is ibm for example I want to render an error of not supported provider. 

So I want to add the `ProviderTypeAll` case and we can change the naming to  `ProviderTypeUnsupported`

Reply:
Why do we want to return all sources? I would agree, that for such a case `ProviderTypeAll` makes sense, but do we have a usecase for all sources?

Reply:
Yes, when the provider is not mentioned we are listing all the sources. 
as you can see in `sources_service.go`

Reply:
but there it's already solved, do we need to fix it in this PR? sounds like the change is controversial and not really needed in given PR, in such cases, I'm up for extracting it, so it can be discussed properly separately :)

Reply:
I added the switch case for few reasons: 

1. Introducing a new switch case for `ProviderTypeAll` aligns with the existing pattern of switch cases we have in the reservations service. 

2. The `ListSources` function is more  readable and cleaner.

3. separation of logic. The `ListSources`  function was previously handling two different logics within a single, lengthy function. By isolating the `ProviderTypeAll` case, we can clearly differentiate the distinct behaviors. 

4. The need to access` r.Path` information for rendering accurate metadata inside the `ListProvisioningX` function in the `sources_client.go` files. I did not want to add another parameter to the `ListProvisioningX` functions related to the endpoint path. I thought we should wrap the client function with a wrapper function and get the r and w parameters.

Reply:
And moreover, the "" provider was not provided is not the same case as "ibm".